FirebaseSessions/Tests/Unit/FirebaseSessionsTests+BaseBehaviors.swift

```diff
@@ -0,0 +1,148 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+
+#if SWIFT_PACKAGE
+  import FirebaseSessionsObjC
+#endif // SWIFT_PACKAGE
+
+@testable import FirebaseSessions
+
+final class FirebaseSessionsTests_BaseBehaviors: FirebaseSessionsTests {
+  // MARK: - Test Settings & Sampling
+
+  func test_settingsDisabled_doesNotLogSessionEventButDoesFetchSettings() {
```

Comment:
These are such powerful tests to validate the behavior of the library in whole. Great work here!